libs/coin-framework/src/account/helpers.ts

```diff
@@ -0,0 +1,414 @@
+  //   (copy as BitcoinAccount).bitcoinResources = initialBitcoinResourcesValue;
+  // }
+  delete copy.nfts;
+  return copy as T;
+}
+
+export function findSubAccountById(
+  account: Account,
+  id: string
+): SubAccount | null | undefined {
+  return (account.subAccounts || []).find((a) => a.id === id);
+}
+
+// get the token accounts of an account, ignoring those that are zero IF user don't want them
+export function listSubAccounts(account: Account): SubAccount[] {
+  const accounts = account.subAccounts || [];
+
+  if (getEnv("HIDE_EMPTY_TOKEN_ACCOUNTS")) {
+    return accounts.filter((a) => !a.balance.isZero());
+  }
+
+  return accounts;
+}
+
+export type FlattenAccountsOptions = {
+  enforceHideEmptySubAccounts?: boolean;
+};
+
+export function flattenAccounts(
+  topAccounts: AccountLikeArray,
+  o: FlattenAccountsOptions = {}
+): AccountLike[] {
+  const accounts: AccountLike[] = [];
+
+  for (let i = 0; i < topAccounts.length; i++) {
+    const account = topAccounts[i];
+    accounts.push(account);
+
+    if (account.type === "Account") {
+      const subAccounts = o.enforceHideEmptySubAccounts
+        ? listSubAccounts(account)
+        : account.subAccounts || [];
+
+      for (let j = 0; j < subAccounts.length; j++) {
+        accounts.push(subAccounts[j]);
+      }
+    }
+  }
+
+  return accounts;
+}
+
+export const shortAddressPreview = (addr: string, target = 20): string => {
+  const slice = Math.floor((target - 3) / 2);
+  return addr.length < target - 3
+    ? addr
+    : `${addr.slice(0, slice)}...${addr.slice(addr.length - slice)}`;
+};
+
+export const isAccountBalanceUnconfirmed = (account: AccountLike): boolean =>
+  account.pendingOperations.some(
+    (op) => !account.operations.find((o) => o.hash === op.hash)
+  ) || account.operations.some((op) => !op.blockHeight);
+
+export const isUpToDateAccount = (account: Account | null | undefined) => {
+  if (!account) return true;
+  const { lastSyncDate, currency } = account;
+  const { blockAvgTime } = currency;
+  if (!blockAvgTime) return true;
+  const outdated =
+    // FIXME: same here, we need to use valueOf for typescript to compare dates
+    Date.now().valueOf() - (lastSyncDate.valueOf() || 0) >
+    blockAvgTime * 1000 + getEnv("SYNC_OUTDATED_CONSIDERED_DELAY");
+  return !outdated;
+};
+
+export const getVotesCount = (
+  account: AccountLike,
+  parentAccount?: Account | null | undefined
+): number => {
+  // const mainAccount = getMainAccount(account, parentAccount);
+
+  // FIXME find a way to make it per family?
+  // switch (mainAccount.currency.family) {
+  //   case "tezos":
+  //     return isAccountDelegating(account) ? 1 : 0;
+  //   case "tron":
+  //     return (mainAccount as TronAccount)?.tronResources?.votes.length || 0;
+  //   case "osmosis":
+  //   case "cosmos":
+  //     return (
+  //       (mainAccount as CosmosAccount)?.cosmosResources?.delegations.length || 0
+  //     );
+  //   default:
+  //     return 0;
+  // }
+  return 0;
```

Comment:
this is a regression for analytics

Reply:
I remove this method from *coin-framework*. It seems not to be useful in this package for the time being.